graphqshell/src/GraphQL/Client.hs

```diff
@@ -0,0 +1,55 @@
+module GraphQL.Client
+  (
+   Client
+  , ClientError (..)
+  , clientEndpoint
+  , mkClient
+  , runRequest
+   ) where
+import Relude hiding (Option)
+import qualified Data.Aeson as J
+import Network.HTTP.Req
+import qualified Text.URI as URI
+import GraphQL.Types
+import Control.Monad.Catch (MonadThrow)
+import Control.Exception.Safe (throw)
+
+data Client = Client {
```

Comment:
Same question: let's package that up into a monad and thus provide a way to run it in a differnt implementation in tests.